fix: Refactor AudioRecordingController by transfering all the business logic

evgenysavonkin opened the pull request:
Refactor AudioRecordingController by transfering all the business logic into ServiceAudioRecordingsImpl class

evgenysavonkin committed:
fix: Refactor AudioRecordingController by transfering all the business logic into ServiceAudioRecordingsImpl class

Wooodward commented on back/src/main/java/com/music_streaming_app/service/impl/ServiceAudioRecordingsImpl.java:
```diff
@@ -54,4 +61,50 @@ public boolean saveAudioRecording(DtoAudioRecording dtoAudioRecording) {
+        List<DtoAudioRecording> dtoAudioRecordings = new ArrayList<>();
+
+        for (AudioRecording audioRecording : getAllAudioRecordings()) {
+            DtoAudioRecording dtoAudioRecording = DtoAudioRecording.builder()
```
в контроллере остался метод toDto, здесь нужно использовать его

Wooodward commented on back/src/main/java/com/music_streaming_app/service/impl/ServiceAudioRecordingsImpl.java:
```diff
@@ -54,4 +61,50 @@ public boolean saveAudioRecording(DtoAudioRecording dtoAudioRecording) {
+    public ResponseEntity<List<DtoAudioRecording>> getAllDtoAudioRecordings() {
+        List<DtoAudioRecording> dtoAudioRecordings = new ArrayList<>();
+
+        for (AudioRecording audioRecording : getAllAudioRecordings()) {
```
нужно переписать с использованием stream api

Wooodward commented on back/src/main/java/com/music_streaming_app/service/impl/ServiceAudioRecordingsImpl.java:
```diff
@@ -54,4 +61,50 @@ public boolean saveAudioRecording(DtoAudioRecording dtoAudioRecording) {
+
+    public ResponseEntity<StreamingResponseBody> getAudioRecording(Optional<AudioRecording> audioRecordingOptional) {
+        Blob audioBlob = audioRecordingOptional.get().getAudioBlob();
+        StreamingResponseBody responseBody = outputStream -> {
```
вот это можно вынести в отдельный метод с названием типа toStreamingResponseBody
взять этот новый метод вместе с методом toDto из контроллера и вынести в отдельный класс AudioRecordingConverter в отдельном пакете converter

Wooodward commented on back/src/main/java/com/music_streaming_app/service/impl/ServiceAudioRecordingsImpl.java:
```diff
@@ -44,6 +50,7 @@ public boolean saveAudioRecording(DtoAudioRecording dtoAudioRecording) {
             audioRecording = AudioRecording.builder()
                     .audioBlob(new SerialBlob(dtoAudioRecording.getFile().getBytes()))
+                    .artist(dtoAudioRecording.getArtist())
 //                    .author(dtoAudioRecording.getArtist())
```
такого поля в классе нет
комментарий надо или выпилить, или поменять поле на существующее и раскомментить

Wooodward commented on back/src/main/java/com/music_streaming_app/service/impl/ServiceAudioRecordingsImpl.java:
```diff
@@ -44,6 +50,7 @@ public boolean saveAudioRecording(DtoAudioRecording dtoAudioRecording) {
         try {
             audioRecording = AudioRecording.builder()
```
надо это все унести в класс-конвертер

Wooodward commented on back/src/main/java/com/music_streaming_app/service/impl/ServiceAudioRecordingsImpl.java:
```diff
@@ -5,12 +5,18 @@
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 import java.util.Set;
```
из методов не нужно возвращать Optional
из них должен идти уже результат
вариант переделать репозиторий, чтобы вместо объекта возвращался опшинал с объектом, хибернейт так умеет
но, лучше просто выпилить опшинал, и просто добавить проверку на null  после получения из базы

Wooodward replied:
аннотация transactional здесь лишняя, надо убрать

Wooodward commented on back/src/main/java/com/music_streaming_app/service/impl/ServiceAudioRecordingsImpl.java:
```diff
@@ -54,4 +61,50 @@ public boolean saveAudioRecording(DtoAudioRecording dtoAudioRecording) {
+            }
+        };
+
+        HttpHeaders headers = new HttpHeaders();
```
заполнение хедеров имеет смысл вынести в отдельный метод

Wooodward commented on back/src/main/java/com/music_streaming_app/service/impl/ServiceAudioRecordingsImpl.java:
```diff
@@ -54,4 +61,50 @@ public boolean saveAudioRecording(DtoAudioRecording dtoAudioRecording) {
         return true;
     }
+
+    public ResponseEntity<StreamingResponseBody> getAudioRecording(Optional<AudioRecording> audioRecordingOptional) {
```
название метода неправильное, никакого гета здесь не происходит
здесь объект одного типа конвертится в другой 
если этот метод используется только вместе с getById, то логика должна лежать в getById

Wooodward commented on back/src/main/java/com/music_streaming_app/controller/AudioRecordingController.java:
```diff
@@ -30,7 +23,6 @@ public class AudioRecordingController {
 
     private final ServiceAudioRecordingsImpl serviceAudioRecordingsImpl;
```
всегда используем тип абстракции, а не реализации
в данном случае - ServiceAudioRecordings
спринг сам инжектнет нужного кандидата

Wooodward commented on back/src/main/java/com/music_streaming_app/controller/AudioRecordingController.java:
```diff
@@ -40,70 +32,22 @@ public ResponseEntity<String> saveAudioRecording(@ModelAttribute DtoAudioRecordi
 
-    // Получаем все записи из БД
     @GetMapping("/all")
     public ResponseEntity<List<DtoAudioRecording>> getAllAudioRecordings() {
```
нужно добавить логгирование в каждый метод типа 
log.info("methodName -> argument")

evgenysavonkin committed:
fix: Refactor AudioRecordingController by transfering all the business logic classes
1. Create class AudioRecordingConverter with its static methods
2. Add new methods to ServiceAudioRecordings interface and inject its implementation through abstraction in AudioRecordingController

evgenysavonkin committed:
fix: Refactor AudioRecordingController by transfering all the business logic classes
1. Remove @Transaction on saveAudioRecoding method in ServiceAudioRecordingsImpl class

Wooodward commented on back/src/main/java/com/music_streaming_app/controller/AudioRecordingController.java:
```diff
@@ -28,102 +21,35 @@ public class AudioRecordingController {
 
         logger.info("Post dto: " + dtoAudioRecording.toString());
-        boolean saveInDateBase = serviceAudioRecordingsImpl.saveAudioRecording(dtoAudioRecording);
+        boolean saveInDateBase = serviceAudioRecordings.saveAudioRecording(dtoAudioRecording);
```
вот этот буль, который здесь возвращается, по сути вообще бесполезный и нам ни о чем не скажет
надо либо переделывать реализацию, чтобы возвращалось хоть что-то осмысленное, что мы фронту можем отдать типа id
либо второй вариант, не сохранять результат ни в каких переменных, а фронту сразу отдавать success с одинаковым сообщением
ну т.е. мы убираем переменную, и пишем в ok("Audio recording is saved succefully") или типо того

Wooodward commented on back/src/main/java/com/music_streaming_app/controller/AudioRecordingController.java:
```diff
@@ -28,102 +21,35 @@ public class AudioRecordingController {
 
         return ResponseEntity.ok(
-                serviceAudioRecordingsImpl.getFeaturedAudioRecordings()
+                serviceAudioRecordings.getFeaturedAudioRecordings()
```
вот весь этот стрим должен быть в сервисе, а сюда только результат приходить

Wooodward commented on back/src/main/java/com/music_streaming_app/service/impl/ServiceAudioRecordingsImpl.java:
```diff
@@ -33,25 +38,40 @@ public Set<AudioRecording> getFeaturedAudioRecordings() {
-        AudioRecording audioRecording;
+        AudioRecording audioRecording = AudioRecordingConverter.toAudioRecording(dtoAudioRecording);
+        repositoryAudioRecordings.save(audioRecording);
+        return true;
```
я в контроллере написал, что возвращать отсюда булевое значение, тем более захардкоженное - такая себе идея, лучше или метод войдовым сделать или вернуть что-то осмысленное типа id или name или что-то такое, чтобы можно было понять, что сохранили
тут на твое усмотрение, сделать метод войдовым и не греть себе голову - доступная опция

Wooodward commented on back/src/main/java/com/music_streaming_app/service/impl/ServiceAudioRecordingsImpl.java:
```diff
@@ -33,25 +38,40 @@ public Set<AudioRecording> getFeaturedAudioRecordings() {
-        repositoryAudioRecordings.save(audioRecording);
-        return true;
+    @Override
+    public ResponseEntity<List<DtoAudioRecording>> getAllDtoAudioRecordings() {
```
проведи вертикальное форматирование кода, все public - наверх, private - вниз

evgenysavonkin committed:
fix: Refactor AudioRecordingController/transfer logic into services
1. Change "saveAudioRecording" method in AudioRecordingController to return only "Audio recording is saved successfully" message
2. Change signatures of "getFeaturedAudioRecordings" and "saveAudioRecording" methods in ServiceAudioRecordings
3. Change "getFeaturedAudioRecordings" method in ServiceAudioRecordingsImpl to return ResponseEntity.ok(List<DtoAudioRecordings>)
